src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -123,28 +131,61 @@ public static void loadStyleMaps(final Path basePath) throws IOException
             {
                 final Path path = it.next();
 
-                final String style = path.getParent().getFileName().toString();
-
-                //Don't treat generic schematics as decorations or huts - ex: supply ship
-                if (NULL_STYLE.equals(style) || MINER_STYLE.equals(style))
-                {
-                    continue;
-                }
-
                 if (path.toString().endsWith(SCHEMATIC_EXTENSION))
                 {
+                    String style = "";
+                    String schematicCategory = "";
+                    if (path.getParent().toString().startsWith(basePath.toString()))
+                    {
+                        style = path.getParent().toString().substring(basePath.toString().length());
+                        if (style.startsWith("/"))
+                        {
+                            style = style.substring(1);
+                        }
+                        final int indexSeparator = style.indexOf('/');
+                        if (indexSeparator!=-1)
+                        {
+                            schematicCategory = style.substring(0,indexSeparator);
+                            style = style.substring(indexSeparator+1);
+                        }
+                    }
+
+                    //Don't treat generic schematics as decorations or huts - ex: supply ship
+                    if (NULL_STYLE.equals(style) || MINER_STYLE.equals(style))
+                    {
+                        continue;
+                    }
+
                     final String filename = path.getFileName().toString().split("\\.nbt")[0];
                     final String hut = filename.split("\\d+")[0];
 
-                    if (isSchematicHut(hut))
+                    if (SCHEMATICS_HUTS.equals(schematicCategory) && isSchematicHut(hut))
                     {
                         addHutStyle(hut, style);
                         incrementHutMaxLevel(hut);
                     }
-                    else
+                    else if (SCHEMATICS_HUTS.equals(schematicCategory) || SCHEMATICS_DECORATIONS.equals(schematicCategory))
                     {
+                        if (isSchematicHut(hut))
+                        {
+                            Log.getLogger().warn(path + " look like a hut but will not be considered as a hut!");
```

Comment:
This happen when a decoration schematic filename match a hut but is in decorations folder.
I added this warning to the log because the user might have put a hut schematic in the decorations folder without realising that it won't be working as a hut (but he will can still be able to build it).